### PR TITLE
Ex 111 internet explorer incompatibility

### PR DESCRIPTION
**Pull request by KellyInnovation**
These are the major fixes for Internet Explorer functionality.

### CODE REVIEW

File: lfucg-exactions/client/js/reducers/activeFormReducer.js [hunk 1]
```diff
@@ -12,14 +12,19 @@ export default function activeFormReducer(state = {}, action) {
...
-        map((entry) => {
-            if (entry[1] && entry[0].startsWith('filter')) {
-                delete state[entry[0]];
+        map((key_name) => {
+            if (key_name.indexOf('filter') !== -1) {
+                delete state[key_name];
             }
-        })(Object.entries(state));
+        })(Object.keys(state));
+        // map((entry) => {
```
Comment (Wei-KeatKong) [inline lfucg-exactions/client/js/reducers/activeFormReducer.js]: remove if not needed

File: lfucg-exactions/client/js/reducers/tokenReducer.js [hunk 2]
```diff
@@ -11,8 +11,10 @@ const tokenReducer = (state = {}, action) => {
     switch (endpoint) {
     case LOGIN:
         try {
+            console.log('TOKEN LOGIN', action.response);
```
Comment (Wei-KeatKong) [inline lfucg-exactions/client/js/reducers/tokenReducer.js]: remove console logs

File: lfucg-exactions/client/webpack.config.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module.exports = {
   entry: {
     // 'main': './js/main.js',
-    'dashboard': './js/dashboard.js',
+    // 'dashboard': './js/dashboard.js',
```
Comment (Wei-KeatKong) [inline lfucg-exactions/client/webpack.config.js]: remove commented line

File: lfucg-exactions/server/api/serializers.py [hunk 4]
```diff
@@ -1,41 +1,41 @@
...
-            'username',
-            'email',
-            'first_name',
-            'last_name',
-            'is_superuser',
-            'token',
-            'permissions',
-            'profile',
-        )
+# from django.contrib.auth.models import User
```
Comment (Wei-KeatKong) [inline lfucg-exactions/server/api/serializers.py]: remove commented?

File: lfucg-exactions/server/api/views.py [hunk 5]
```diff
@@ -1,11 +1,11 @@
-from rest_framework.generics import RetrieveAPIView
+# from rest_framework.generics import RetrieveAPIView
```
Comment (Wei-KeatKong) [inline lfucg-exactions/server/api/views.py]: remove commented?

File: lfucg-exactions/client/js/actions/apiActions.js [hunk 6]
```diff
@@ -1372,12 +1372,12 @@ export function searchQuery(isCSV) {
 
             const queryString = compose(
                 reduce((acc, value) => acc + value, query_all),
-                map((a) => {
-                    const field = a[0].slice(7, a[0].length);
-                    return `&${field}=${a[1]}`;
+                map((key_name) => {
+                    const field = key_name.slice(7, key_name.length);
```
Comment (Wei-KeatKong) [inline lfucg-exactions/client/js/actions/apiActions.js]: what's this magic number 7?

File: lfucg-exactions/client/js/components/AccountLedgerForm.js [hunk 7]
```diff
@@ -474,7 +486,7 @@ function mapDispatchToProps(dispatch, params) {
                         [field_show]: value,
                     };
                     dispatch(formUpdate(account_update));
-                    if (field === 'account_from' && !value_name.includes('LFUCG')) {
+                    if (field === 'account_from' && (value_name.indexOf('LFUCG') === -1)) {
```
Comment (Ellun) [inline lfucg-exactions/client/js/components/AccountLedgerForm.js]: Just curious what the benefit of switching from `includes` to `indexOf` is?

Comment (KellyInnovation) [reply lfucg-exactions/client/js/components/AccountLedgerForm.js]: includes does not work on Internet Explorer, indexOf does.

File: lfucg-exactions/client/js/actions/apiActions.js (hunk 6 above)
Comment (KellyInnovation) [reply lfucg-exactions/client/js/actions/apiActions.js]: This is for filter

**Comment by JMStewart00** [comment]
@KellyInnovation - I didn't get to look at the substr stuff but the check on prev needs to be checking prev instead of next.substr ... does that make sense?

**Pull request merged**